public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -140,20 +140,17 @@ class MetricsPanelCtrl extends PanelCtrl {
         }
       }
 
-      if (this.dataFormat & PanelDataFormat.Legacy) {
-        // The result should already be processed, but just in case
-        if (!data.legacy) {
-          data.legacy = data.series.map(v => {
-            if (isDataFrame(v)) {
-              return toLegacyResponseData(v);
-            }
-            return v;
-          });
-        }
+      if (this.useDataFrames) {
```

Comment:
missing a ! negation :)

Reply:
I have never once made an error like this before.  My self worth is totally in question now. ;)